craft_parts/plugins/rust_plugin.py

```diff
@@ -111,32 +153,109 @@ def get_build_snaps(self) -> Set[str]:
     @override
     def get_build_packages(self) -> Set[str]:
         """Return a set of required packages to install in the build environment."""
-        return {"gcc", "git"}
+        return {"curl", "gcc", "git", "pkg-config", "findutils"}
+
+    def _check_system_rust(self) -> bool:
+        """Check if Rust is installed on the system."""
+        try:
+            rust_version = subprocess.check_output(["rustc", "--version"])
+            cargo_version = subprocess.check_output(["cargo", "--version"])
+            return "rustc" in rust_version.decode(
+                "utf-8"
+            ) and "cargo" in cargo_version.decode("utf-8")
+        except (subprocess.CalledProcessError, FileNotFoundError):
+            return False
+
+    def _check_rustup(self) -> bool:
+        try:
+            rustup_version = subprocess.check_output(["rustup", "--version"])
+            return "rustup" in rustup_version.decode("utf-8")
+        except (subprocess.CalledProcessError, FileNotFoundError):
+            return False
+
+    def _get_setup_rustup(self, channel: str) -> List[str]:
+        return [
+            f"""\
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
+sh -s -- -y --no-modify-path --profile=minimal --default-toolchain {channel}
+"""
+        ]
 
     @override
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
-        return {"PATH": "${HOME}/.cargo/bin:${PATH}"}
+        return {
+            "PATH": "${HOME}/.cargo/bin:${PATH}",
+        }
+
+    @override
+    def get_pull_commands(self) -> List[str]:
+        """Return a list of commands to run during the pull step."""
+        options = cast(RustPluginProperties, self._options)
+        if not options.rust_channel and self._check_system_rust():
+            logger.info("Rust is installed on the system, skipping rustup")
+            return []
+
+        rust_channel = options.rust_channel or "stable"
+        if rust_channel == "none":
```

Comment:
Is "none" a valid channel value or something used for debug? It's not mentioned in the docstring above.